tango_simlib/tests/test_fandango_json_parser.py

```diff
@@ -0,0 +1,183 @@
+"""This module tests the fandango_json_parser script.
+"""
+import unittest
+import logging
+import pkg_resources
+
+import tango
+
+from tango_simlib.utilities import fandango_json_parser
+
+MODULE_LOGGER = logging.getLogger(__name__)
+
+
+EXPECTED_DEVICE_PARAMETERS = [
+    'commands', 'class_properties', 'name',
+    'level', 'server', 'host', 'dev_class',
+    'attributes', 'properties'
+    ]
+
+EXPECTED_CMD_PROPERTY_PARAMETERS = [
+    'name', 'dtype_in',
+    'doc_in', 'dtype_out',
+    'doc_out'
+    ]
+
+# The desired information for the attribute 'Status' when the database2
+# json file is parsed by the FandangoExportDeviceParser.
+EXPECTED_STATUS_ATTR_INFO = {
+    'alarms': {
+        'delta_t': 'Not specified',
+        'delta_val': 'Not specified',
+        'extensions': '[]',
+        'max_alarm': 'Not specified',
+        'max_warning': 'Not specified',
+        'min_alarm': 'Not specified',
+        'min_warning': 'Not specified'
+        },
+    'color': 'Lime',
+    'data_format': tango._tango.AttrDataFormat.SCALAR,
+    'data_type': tango._tango.CmdArgType.DevString,
+    'database': 'monctl:10000',
+    'description': '',
+    'device': 'sys/database/2',
+    'display_unit': 'No display unit',
+    'events': {
+        'arch_event': {
+            'archive_abs_change': 'Not specified',
+            'archive_period': 'Not specified',
+            'archive_rel_change': 'Not specified',
+            'extensions': '[]'
+            },
+        'ch_event': {
+            'abs_change': 'Not specified',
+            'extensions': '[]',
+            'rel_change': 'Not specified'
+            },
+        'per_event': {
+            'extensions': '[]',
+            'period': '1000'
+            }
+        },
+    'format': '%s',
+    'label': 'Status',
+    'max_alarm': 'Not specified',
+    'min_alarm': 'Not specified',
+    'model': 'monctl:10000/sys/database/2/Status',
+    'name': 'Status',
+    'polling': 0,
+    'quality': 'ATTR_VALID',
+    'standard_unit': 'No standard unit',
+    'string': 'Device is OK',
+    'time': 1521724934.799581,
+    'unit': '',
+    'value': 'Device is OK',
+    'writable': 'READ'
+    }
+
+
+class GenericSetup(unittest.TestCase):
```

Comment:
Minor, but we are only using this base class once, so not really necessary.  The SimDD tests, where it came from, use it a few times.  You don't need to change it.

Reply:
OK